catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/operation/impl/QueryResponseImpl.java

```diff
@@ -302,7 +302,7 @@ private Result handleTake() {
         Result result = null;
         try {
             result = queue.take();
-            if (result == poisonPillResult) {
+            if (result == POISON_PILL_RESULT) {
```

Comment:
Well, I'm not sure the profiler would help. It looks like we always call .take() until we get back a null. However, if someone created a new strategy that took (just for example) the first 10 results of any query response and then tossed the rest without iterating through it, you'd have a memory leak.